helix-core/src/main/java/org/apache/helix/model/InstanceConfig.java

```diff
@@ -302,13 +310,14 @@ public void setInstanceDisabledReason(String disabledReason) {
    * It will be a no-op when instance is enabled.
    */
   public void setInstanceDisabledType(InstanceConstants.InstanceDisabledType disabledType) {
```

Comment:
Shall we make it more generic? Not only for disable but evacuate and else?

Probably call "setInstanceNonServingType" ?

Reply:
I agree that we need to build a more generic API for more than just the DISABLE InstanceOperation. We will open a future PR with this change, as the disable type, disable reason, etc. needs to be rethought completely.

Reply:
@GrantPSpencer, Please factor this into the design initially discussed for disable reason.